README.md

```diff
@@ -1,2 +1,82 @@
 # build-and-publish-to-pypi
-A GitHub Actions workflow for building a Python package and publishing it to PyPI.
+
+This is a [composite GitHub Action](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action) for building a pure-Python package and publishing it to the [Python Package Index (PyPI)](https://pypi.org).
+
+This actions rolls into a single step two tasks that are commonly run together:
+
+1. Build a Python package's distribution, accomplished with the PyPA's [build](https://pypa-build.readthedocs.io/en/stable/index.html) tool.
+2. Upload the distribution to PyPI, using the [pypa/gh-action-pypi-publish](https://github.com/pypa/gh-action-pypi-publish) action.
+
+Since this action uses the [build](https://pypa-build.readthedocs.io/en/stable/index.html) tool, any [PEP 517](https://peps.python.org/pep-0517/)-compatible packaging tool is supported, including [setuptools](https://setuptools.pypa.io/en/latest/).
+You can make your project PEP 517-compatible by adding a `project.toml` file and specifying the build backend in a `[build-system]` section ([see setuptools' tutorial on this](https://setuptools.pypa.io/en/latest/build_meta.html)).
+
+**However, building wheels for multiple platforms is not supported by this action.**
+If your package compiles extensions, you'll need to build your own multi-platform GitHub Actions job that builds a wheel on each platform.
+
+## Example usage
+
+```yaml
+name: Python CI
+
+"on":
+  push:
+    tags:
+      - "*"
+  pull_request: {}
+
+jobs:
+
+  pypi:
+
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/')
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # full history for setuptools_scm
+
+      - name: Build and publish
+        uses: lsst-sqre/build-and-publish-to-pypi@v1
+        with:
+          pypi-token: ${{ secrets.PYPI_SQRE_ADMIN }}
+```
+
+## Inputs
+
+- `pypi-token` (string, required) an API token for PyPI.
+- `python-version` (string, optional) the Python version. Default is `3.10`.
```

Comment:
Thinking ahead here, I wonder if it would make sense to always require specifying the version. My intuition is that I would normally want the build and PyPI upload to be done with the same Python version that the package is targeting (it may not build on newer versions, for instance, or may build binary wheels that should match the supported Python version), and if it's always specified in the package GitHub Actions workflow definition, I'll remember to update it when I update the Python versions for tests, docs (if relevant), etc. But if the default of the composite action is used, I'm likely to forget that it has a default version, or what that version is, and it seems more likely that we'll end up trying to publish with a different version than the package was tested with.

That's more awkward with this action than some others, such as docs, since it only runs on tags, so if it does fail due to a version mismatch one wouldn't know until the release is already tagged and then fixing it would probably require making and tagging a new release.

Reply:
Hm, this is an interesting question. My original thinking was we'd update the Python version in major releases of the action so that dependabot would effectively help us keep up to date by default.

However, I think you're right that we'll always probably need to keep an eye on our Python versions. For example, I think the best way to do a matrix of tox-based tests would be to wrap up tox in a composite action and let the package's workflow then control what versions of Python it cares to test against.